### PR TITLE
Add Summary Status for Facility Lists

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Adds a field to the facility list detail endpoint that is a DISTINCT list of all item statuses in the facility list. This list is inspected in the front-end and is used to show a status message in the pagination row.

Connects #327 

### Demo

Processing:

image

Error:

image

## Testing

* Check out this branch and `server`
* Go to [:6543](http://localhost:6543/) and log in
* Upload a list
  - [ ] Ensure the list details page shows a summary status above the list items in the table
* `manage batch_process` the list
  - [ ] Ensure the list details page shows the correct summary status at each step

### CODE REVIEW

File: src/django/api/serializers.py [hunk 1]
```diff
@@ -171,6 +172,42 @@ def get_items_url(self, facility_list):
...
+            FacilityListItem.CONFIRMED_MATCH,
+        ])
+
+        item_statuses = set(
+            facility_list.facilitylistitem_set
+            .values_list('status', flat=True)
+            .distinct())
+
+        if (item_statuses.intersection(ERROR_STATUSES)):
+            status = 'ERROR'
```
Comment (jwalgran) [inline src/django/api/serializers.py]: A list can have both errors and pending items (many lists we have seen so far have some parse errors on individual lines. On average, the data is pretty dirty). We should consider returning  multiple additional "computed flags" from the serializer, one for the presence of an error and one for "waiting on you or waiting on us" processing status.

Comment (rajadain) [reply src/django/api/serializers.py]: Switched to send a distinct list of all item statuses from the back-end, and have the front-end choose messages accordingly.

File: src/app/src/util/util.js [hunk 2]
```diff
@@ -453,3 +456,29 @@ export const makeFacilityListDataURLs = (id, count) => {
...
+        facilityListItemStatusChoicesEnum.PARSED,
+        facilityListItemStatusChoicesEnum.GEOCODED,
+        facilityListItemStatusChoicesEnum.GEOCODED_NO_RESULTS,
+    ].includes(s)) ?
+        facilityListSummaryStatusMessages.PROCESSING : '';
+    const completeMessage = statuses.every(s => [
+        facilityListItemStatusChoicesEnum.MATCHED,
+        facilityListItemStatusChoicesEnum.CONFIRMED_MATCH,
+    ].includes(s)) ?
+        facilityListSummaryStatusMessages.COMPLETED : '';
```
Comment (kellyi) [inline src/app/src/util/util.js]: Not sure how to suggest cleaning it up, but this series of chained ternaries is a little hard to follow. I think the test does enough to verify that it's working, but in theory we could revise it with something like https://lodash.com/docs/4.17.11#cond that would do a form of pattern matching, which is I think what we want here.

Comment (rajadain) [reply src/app/src/util/util.js]: `cond` does a single match, but here we want multiple matches.

Comment (kellyi) [reply src/app/src/util/util.js]: Now that I'm looking at it again, this could also be a reduce operation that just rolls through the list and figures out what to do on each pass.

This is probably fine to leave as is assuming we're confident that the test is working, which it seems to be!

**Comment by rajadain** [comment]
How does this look:

Black and bold:

image

vs

Black only:

image

vs

Bold only:

image

**Comment by rajadain** [comment]
Added "Bold only" styling in 6762406.

**Comment by kellyi** [comment]
Cool. Bold only looks great!

**Comment by jwalgran** [comment]
Looking at this now

File: src/app/src/util/util.js (hunk 2 above)
Comment (jwalgran) [reply src/app/src/util/util.js]: 👍

**Comment by rajadain** [comment]
> One way of dealing with this would be to re-fetch the list details using the existing API and update the state.

Alright I'll look into this. May be tricky to keep the pagination state in place when refreshing the larger list.

File: src/app/src/__tests__/utils.tests.js [hunk 3]
```diff
@@ -1022,3 +1025,47 @@ it('creates a list of data URLs for retrieving facility list items data', () =>
         smallerExpectedURLs,
     )).toBe(true);
 });
+
+it('creates a summary status message given a list of facility list item statuses', () => {
```
[the remark this answers is not in the document]
Comment (rajadain) [reply src/app/src/__tests__/utils.tests.js]: Added in 1f683e3.

**Comment by rajadain** [comment]
Added ff57c65 which updates the summary status as the user is interacting with the items in the list.

The manner it is implemented in makes the shape of the data from the confirm and reject endpoints different from the other list item details endpoints, but perhaps that is okay for now.

This is ready for another look.

**Comment by jwalgran** [comment]
Looks like a reasonable solution. Testing now

**Comment by rajadain** [comment]
Squashed fixups. Merging now. Thanks for all the feedback!
